web3/main.py

```diff
@@ -511,12 +512,11 @@ async def __aexit__(
         "when instantiating via ``async for``."
     )
     async def __aiter__(self) -> AsyncIterator[Self]:
-        if not await self.provider.is_connected():
```

Comment:
Does this check matter? I can't decide, ha

Reply:
I think we should keep `connect`  idempotent and force a re-connect on each iteration to make sure we are creating a clean slate. Thoughts there?

Reply:
Also, since we're disconnecting after yielding now, I think this makes more sense.

Reply:
okay, cool. That makes sense to me!